### PR TITLE
Bug 1862432: unmask prometheus-adapter service monitor

### PR DESCRIPTION
**Pull request by dgrisonnet**
* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

cc @s-urbaniak

### CODE REVIEW

File: assets/prometheus-adapter/service-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  name: prometheus-adapter
+  namespace: openshift-monitoring
+spec:
+  endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    interval: 30s
+    port: https
+    scheme: https
+    tlsConfig:
+      insecureSkipVerify: true
```
Comment (dgrisonnet) [inline assets/prometheus-adapter/service-monitor.yaml:15]: Should I add the certs in a follow-up PR?

Comment (s-urbaniak) [reply assets/prometheus-adapter/service-monitor.yaml:15]: We have to do it here :-) look for a reference i.e. here: https://github.com/openshift/cluster-monitoring-operator/blob/4b907eac67450edad388fe04f51343f8d6582d6d/jsonnet/prometheus-operator.jsonnet#L104-L119

**Comment by s-urbaniak** [comment]
good initial start :+1 you'll have to deploy the service monitor in CMO's reconcile code, right after the deployment of prometheus-adapter: https://github.com/openshift/cluster-monitoring-operator/blob/f4b5b9b15b2691a1d6499e9587d32c260d5dbcbd/pkg/tasks/prometheusadapter.go#L179-L181

Injecting the server name can be inspired from here: https://github.com/openshift/cluster-monitoring-operator/blob/004b4975992e84460a09cb935a3a6b1ffb009aaf/pkg/manifests/manifests.go#L1764-L1774

**Comment by dgrisonnet** [comment]
@s-urbaniak I added the cert and the service monitor reconciliation. PTAL

File: pkg/manifests/manifests.go [hunk 2]
```diff
@@ -1711,6 +1712,18 @@ func (f *Factory) PrometheusAdapterService() (*v1.Service, error) {
 	return s, nil
 }
 
+func (f *Factory) PrometheusAdapterServiceMonitor() (*monv1.ServiceMonitor, error) {
```
Comment (s-urbaniak) [inline pkg/manifests/manifests.go:1715]: let's add a unit test in `manifests_test.go` too

**Comment by s-urbaniak** [comment]
one nit, else LGTM :+1:

**Comment by s-urbaniak** [comment]
Please rebase as we fixed the failing e2e test.

**Pull request merged**